src/test/spec/initial_dns_seedlist_discovery.rs

```diff
@@ -68,44 +80,66 @@ async fn run() {
 
-        // The spec tests use two fields to compare against for the authentication database. In the
-        // `parsed_options` field, there is a `db` field which is populated with the database
-        // between the `/` and the querystring of the URI, and in the `options` field, there is an
-        // `authSource` field that is populated with whatever the driver should resolve `authSource`
-        // to from both the URI and the TXT options. Because we don't keep track of where we found
-        // the auth source in ClientOptions and the point of testing this behavior in this spec is
-        // to ensure that the right database is chosen based on both the URI and TXT options, we
-        // just determine what that should be and stick that in all of the options parsed from the
-        // spec JSON.
-        let resolved_db = test_file
-            .options
-            .as_ref()
-            .and_then(|opts| opts.auth_source.clone())
-            .or_else(|| {
-                test_file
-                    .parsed_options
-                    .as_ref()
-                    .and_then(|opts| opts.db.clone())
-            });
+        // "txt-record-with-overridden-ssl-option.json" requires SSL be disabled; see DRIVERS-1324.
+        let requires_tls = match test_file.options {
+            Some(ref options) => options.ssl,
+            None => true,
+        };
+        let client = TestClient::new().await;
+        if requires_tls == client.options.tls_options().is_some()
+            && client.is_replica_set()
+            && client.options.repl_set_name.as_deref() == Some("repl0")
+        {
+            // If the connection URI provides authentication information, manually create the user
+            // before connecting.
+            if let Some(ParsedOptions {
+                user: Some(ref user),
+                password: Some(ref pwd),
+                ref db,
+            }) = test_file.parsed_options
+            {
+                client
+                    .drop_and_create_user(
+                        user,
+                        pwd,
+                        &[],
+                        &[AuthMechanism::ScramSha1, AuthMechanism::ScramSha256],
+                        db.as_deref(),
+                    )
+                    .await
+                    .unwrap();
+            }
+
+            let mut options_with_tls = options.clone();
+            if requires_tls {
+                let tls_options = TlsOptions::builder()
+                    .allow_invalid_certificates(true)
+                    .build();
+                options_with_tls.tls = Some(Tls::Enabled(tls_options));
+            }
+            let mut client = TestClient::with_options(Some(options_with_tls)).await;
+            let mut actual_hosts = client.server_info.hosts.unwrap();
+            if let Some(ref mut arbiters) = client.server_info.arbiters {
+                actual_hosts.append(arbiters);
```

Comment:
localhost:27019 consistently shows up in the arbiters list rather than the host list in the NoTLS variants on CI

Reply:
Looks like that's due to the configs in drivers-evergreen-tools; the basic replica set has [one arbiter](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/3be7d5197facf4f41b56cbda4a346d4e69a8ac62/.evergreen/orchestration/configs/replica_sets/basic.json#L35-L43), but the auth-ssl one [doesn't seem to have any](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/3be7d5197facf4f41b56cbda4a346d4e69a8ac62/.evergreen/orchestration/configs/replica_sets/auth-ssl.json).